app/services/hyrax/collections/collection_member_service.rb

```diff
@@ -67,6 +67,96 @@ def available_member_work_ids
+
+        # Add a work or collection as a member of a collection
+        # @param collection [Hyrax::PcdmCollection] the collection
+        # @param new_member [Hyrax::Resource] the new child collection or child work
+        # @returns [Hyrax::Resource] updated member resource
+        def add_member(collection:, new_member:, user:)
+          message = Hyrax::MultipleMembershipChecker.new(item: new_member).check(collection_ids: collection.id, include_current_members: true)
+          if message
+            new_member.errors.add(:collections, message)
+          else
+            new_member.member_of_collection_ids += [collection.id] # only populate this direction
+            new_member = Hyrax.persister.save(resource: new_member)
+            Hyrax.publisher.publish('object.metadata.updated', object: new_member, user: user)
+          end
+          new_member
+        end
+
+        # Remove collections and/or works from the members set of a collection
+        # @param collection [Hyrax::PcdmCollection] the collection
+        # @param member_ids [Enumerable<Valkyrie::ID>] the ids of the child collections and/or child works to be removed
+        # @returns [Enumerable<Hyrax::Resource>] updated member resources
+        def remove_members_by_ids(collection:, member_ids:, user:)
+          members = Hyrax.query_service.find_many_by_ids(ids: member_ids)
+          remove_members(collection: collection, members: members, user: user)
+        end
+
+        # Remove collections and/or works from the members set of a collection
+        # @param collection [Hyrax::PcdmCollection] the collection
+        # @param members [Enumerable<Valkyrie::Resource>] the child collections and/or child works to be removed
+        # @returns [Enumerable<Hyrax::Resource>] updated member resources
+        def remove_members(collection:, members:, user:)
+          members.map { |member| remove_member(collection: collection, member: member, user: user) }
+        end
+
+        # Remove collections and/or works from the members set of a collection
+        # @param collection [Hyrax::PcdmCollection] the collection
+        # @param member_id [Valkyrie::ID] the id of the child collection or child work to be removed
+        # @returns [Hyrax::Resource] updated member resource
+        def remove_member_by_id(collection:, member_id:, user:)
+          member = Hyrax.query_service.find_by(id: member_id)
+          remove_member(collection: collection, member: member, user: user)
+        end
+
+        # Remove a collection or work from the members set of a collection, also removing the inverse relationship
+        # @param collection [Hyrax::PcdmCollection] the collection
+        # @param member [Hyrax::Resource] the child collection or child work to be removed
+        # @returns [Hyrax::Resource] updated member resource
+        def remove_member(collection:, member:, user:)
+          return collection unless member?(collection: collection, member: member)
```

Comment:
we return the `member` in the method's primary code path, so i think we should do the same for this guard condition.

```suggestion
          return member unless member?(collection: collection, member: member)
```

Reply:
Good catch.  When we were setting the relationship on the collection, it returned collection.  It should have been updated to member when the relationship now that the relationship is only set on the member.  I'll double check all the returns.